src/actions/notes.js

```diff
@@ -13,49 +14,63 @@ export const CREATE_NOTE_START = 'CREATE_NOTE_START';
 export const CREATE_NOTE_SUCCESS = 'CREATE_NOTE_SUCCESS';
 export const CREATE_NOTE_FAILURE = 'CREATE_NOTE_FAILURE';
 
-const postUpdateNote = (bookmarkId, noteId, noteFields) => ({
-  [CALL_API]: {
-    types: [
-        { type: UPDATE_NOTE_START, meta: { bookmarkId, noteId } },
-        { type: UPDATE_NOTE_SUCCESS, meta: { bookmarkId, noteId } },
-        { type: UPDATE_NOTE_FAILURE, meta: { bookmarkId, noteId } },
-    ],
-    method: 'POST',
-    body: JSON.stringify(noteFields),
-    endpoint: `${API_VERSION}/notes/${noteId}`,
-  },
-});
-
-const postDeleteNote = (bookmarkId, noteId) => ({
-  [CALL_API]: {
-    types: [
-        { type: DELETE_NOTE_START, meta: { bookmarkId, noteId } },
-        { type: DELETE_NOTE_SUCCESS, meta: { bookmarkId, noteId } },
-        { type: DELETE_NOTE_FAILURE, meta: { bookmarkId, noteId } },
-    ],
-    method: 'DELETE',
-    endpoint: `${API_VERSION}/notes/${noteId}`,
-  },
-});
-
-const postCreateNote = (bookmarkId, noteFields) => ({
-  [CALL_API]: {
-    types: [
-        { type: CREATE_NOTE_START, meta: { bookmarkId } },
-        { type: CREATE_NOTE_SUCCESS, meta: { bookmarkId } },
-        { type: CREATE_NOTE_FAILURE, meta: { bookmarkId } },
-    ],
-    method: 'POST',
-    body: JSON.stringify(noteFields),
-    endpoint: `${API_VERSION}/bookmarks/${bookmarkId}/notes`,
-  },
-});
-
-export const updateNote = (bookmarkId, noteId, noteFields) => (dispatch) =>
-    dispatch(postUpdateNote(bookmarkId, noteId, noteFields));
+
+const postCreateNote = (bookmarkId, { text, tmpId }) => {
+  const meta = { bookmarkId, note: { text, tmpId } };
+
+  return {
+    [CALL_API]: {
+      types: [
+        { type: CREATE_NOTE_START, meta },
+        { type: CREATE_NOTE_SUCCESS, meta },
+        { type: CREATE_NOTE_FAILURE, meta },
+      ],
+      method: 'POST',
+      body: JSON.stringify({ text }),
+      endpoint: `${API_VERSION}/bookmarks/${bookmarkId}/notes`,
+    },
+  };
+};
+
+const postDeleteNote = (bookmarkId, { noteId, text, noteIndex }) => {
+  const meta = { bookmarkId, note: { id: noteId, text, index: noteIndex } };
+
+  return {
+    [CALL_API]: {
+      types: [
+        { type: DELETE_NOTE_START, meta },
+        { type: DELETE_NOTE_SUCCESS, meta },
+        { type: DELETE_NOTE_FAILURE, meta },
+      ],
+      method: 'DELETE',
+      endpoint: `${API_VERSION}/notes/${noteId}`,
+    },
+  };
+};
+
+const postUpdateNote = (bookmarkId, { noteId, newText, text }) => {
```

Comment:
Same here: `note` and `newFields` (or `newNote`, `updateFields`, you pick)

Reply:
I prefer to have explicit parameters.